sdk/cosmos/azure-cosmos-encryption/src/samples/java/com/azure/cosmos/Program.java

```diff
@@ -46,83 +55,72 @@
     }
 
     private static final String databaseId = "samples";
-    private static final String containerId = "encryptedData";
-    private static final String keyContainerId = "keyContainer";
+    private static final String containerId = "encryptedContainer";
     private static final String dataEncryptionKeyId = "theDataEncryptionKey";
-    private static CosmosAsyncClient client = null;
-    private static EncryptionCosmosAsyncContainer containerWithEncryption = null;
+    private static EncryptionCosmosAsyncClient encryptionCosmosAsyncClient = null;
+    private static EncryptionCosmosAsyncDatabase encryptionCosmosAsyncDatabase = null;
+    private static EncryptionCosmosAsyncContainer encryptionCosmosAsyncContainer = null;
 
     public static void main(String[] args) throws Exception {
         try {
-            // Read the Cosmos endpointUrl and masterKey from configuration.
-            // These values are available from the Azure Management Portal on the Cosmos Account Blade under "Keys".
-            // Keep these values in a safe and secure location. Together they provide administrative access to your
-            // Cosmos account.
+            //NOTE: Please provide credential information in src/samples/resources/settings.properties
             Properties configuration = args.length > 0 ? loadConfig(args[0]) : loadConfig();
-            Program.client = Program.createClientInstance(configuration);
-            Program.initialize(client, configuration);
-            Program.runDemo(client);
+            Program.encryptionCosmosAsyncClient = Program.createClientInstance(configuration);
+            Program.initialize(encryptionCosmosAsyncClient, configuration);
+            Program.runDemo(encryptionCosmosAsyncContainer);
         } catch (CosmosException cre) {
             System.out.println(cre.toString());
         } catch (Exception e) {
             e.printStackTrace();
         } finally {
-            System.out.println("End of demo, press any key to exit.");
+            System.out.println("End of demo, press enter key to exit.");
             System.in.read();
             Program.cleanup();
-            client.close();
+            encryptionCosmosAsyncClient.close();
         }
     }
 
-    private static CosmosAsyncClient createClientInstance(Properties configuration) {
+    private static EncryptionCosmosAsyncClient createClientInstance(Properties configuration) throws MicrosoftDataEncryptionException {
```

Comment:
the fact that we are using the AAP lib is implementation details no?

we should discuss in the meeting for the public api. maybe we shouldn't throw exception from another package and instead we should wrap in CosmosException

Reply:
This is on AzureKeyVaultKeyStoreProvider encryptionKeyStoreProvider = new AzureKeyVaultKeyStoreProvider(tokenCredentials) which will reside on customer code , outside cosmos.